app/modules/utils/rate_limiter.py

```
@@ -0,0 +1,183 @@
+                    await asyncio.sleep(wait_time)
+
+            # Add current timestamp
+            self._request_timestamps.append(now)
+            self.total_requests += 1
+
+    def handle_quota_exceeded(self):
+        """Handle quota exceeded with exponential backoff"""
+        now = datetime.now()
+        if self._last_quota_exceeded:
+            # Calculate exponential backoff with base of 2
+            time_since_last = (now - self._last_quota_exceeded).total_seconds()
```

Ruff (0.8.2)

Local variable `time_since_last` is assigned to but never used

Remove assignment to unused variable `time_since_last`

(F841)